### PR TITLE
add documentation for SqlInlineData attribute

### LINKED ISSUE #273: Add docs for tests using SqlInlineData
Update https://github.com/Azure/azure-functions-sql-extension/blob/main/test/README.md to include a section on how to add new tests including : 

1. What `SqlInlineData` is and why we added it
2. How to add new tests
3. How to make a test not run against a specific language

### PR DESCRIPTION
**Pull request by MaddyDev**
This PR closes #273

### CODE REVIEW

File: test/README.md [hunk 1]
```diff
@@ -24,4 +24,55 @@ Our integration tests are based on functions from the samples project. To run in
      ```
      After the Docker image is running, you just need to set `SA_PASSWORD` environment variable to `{your_password}` and can run tests normally.
      
-     Note: If `SA_PASSWORD` is not set, the tests will assume you're using a local MSSQL installation and default to using integrated auth. MSSQL on Docker does not support integrated auth by default.
\ No newline at end of file
+     Note: If `SA_PASSWORD` is not set, the tests will assume you're using a local MSSQL installation and default to using integrated auth. MSSQL on Docker does not support integrated auth by default.
+
+ ## Adding New Integration Tests
+
+    When addiing a new integration test for a function, make sure to add the function in all supported languages (Under samples\/samples-\<language\> in samples project) unless the test is specific to some language(s). All Integration tests must have atleast a SupportedLanguage input parameter that specifies the language the test runs against.
```
Comment (lucyzhang929) [inline test/README.md:31]: nit: `When adding`
nit: `All integration tests must have a SupportedLanguage`

File: test/README.md [hunk 2]
```diff
@@ -24,4 +24,55 @@ Our integration tests are based on functions from the samples project. To run in
...
+        [Theory]
+        [SqlInlineData()]
+        public async void GetProductsByCostTest(SupportedLanguages lang)
+        {
+            // test code
+            // test assertions
+            // Assert.Equal(expectedResponse, actualResponse, StringComparer.OrdinalIgnoreCase);
+        }
+        ```
+      Ex: When the test has paramters:
```
Comment (lucyzhang929) [inline test/README.md:55]: nit: parameters

File: test/README.md [hunk 3]
```diff
@@ -24,4 +24,55 @@ Our integration tests are based on functions from the samples project. To run in
...
+        [Theory]
+        [SqlInlineData(0, 0)]
+        public async void GetProductsByCostTest(int n, int cost, SupportedLanguages lang)
+        {
+            // test code
+            // test assertions ->
+            // Assert.Equal(expectedResponse, actualResponse, StringComparer.OrdinalIgnoreCase);
+        }
+        ```
+    2. Use [UnsupportedLanguages] attribute over the test to specify the list of languages that the test must not run against.
```
Comment (lucyzhang929) [inline test/README.md:66]: Maybe also add a note about adding comments for why a language is unsupported.

File: test/README.md [hunk 4]
```diff
@@ -24,4 +24,55 @@ Our integration tests are based on functions from the samples project. To run in
...
\ No newline at end of file
+     Note: If `SA_PASSWORD` is not set, the tests will assume you're using a local MSSQL installation and default to using integrated auth. MSSQL on Docker does not support integrated auth by default.
+
+ ## Adding New Integration Tests
+
+    When addiing a new integration test for a function, make sure to add the function in all supported languages (Under samples\/samples-\<language\> in samples project) unless the test is specific to some language(s). All Integration tests must have atleast a SupportedLanguage input parameter that specifies the language the test runs against.
+
+     ### SqlInlineData attribute:
+     
+     SqlInlineData attribute is a custom attribute derived from Xunit Data attribute and it supplies the SupportedLanguage parameter to the test for the test to run against. By default any test decorated with the [SqlInline] attribute will be run against each supported language in the SupportedLanguages enum.
```
Comment (lucyzhang929) [inline test/README.md:35]: nit: `[SqlInlineData]`

File: test/README.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (MaddyDev) [reply test/README.md]: Thanks @Charles-Gagnon

[the remark this answers is not in the document]
Comment (MaddyDev) [reply test/README.md]: Do you mean for me to add a sample function code in all languages?

[the remark this answers is not in the document]
Comment (Charles-Gagnon) [reply test/README.md]: No, just what the test would look like

**Comment by MaddyDev** [comment]
> When I change the view to Rich Diff view, everything after `Adding New Integration Tests` heading seems to be in a code cell.

Weird, I formatted in our ADS and it looked fine, updated it again, looks good now. Thanks for looking into it.

File: test/README.md [hunk 5]
```diff
@@ -24,4 +24,60 @@ Our integration tests are based on functions from the samples project. To run in
...
+        public async void GetProductsByCostTest(int n, int cost, SupportedLanguages lang)
+        {
+            this.StartFunctionHost(nameof(<FUNCTIONNAME>), lang); // Replace <FUNCTIONNAME> with the class name of the function this test is running against
+            // test code here
+        }
+   ```
+
+   ### UnsupportedLanguages attribute:
+
+   UnsupportedLanguages attribute accepts a list of coma separated languages that are not supported for the test. For example, if a test is not relevant to a specific language or if the function is not yet available in the samples-\<language>, we should use UnsupportedLanguages attribute and specify the language(s) for exclusion.
```
Comment (Charles-Gagnon) [inline test/README.md:68]: `comma`

**Review by Charles-Gagnon — APPROVED** [review]

**Review by lucyzhang929 — APPROVED** [review]

**Review by chlafreniere — APPROVED** [review]

**Pull request merged**